app/src/org/commcare/views/notifications/NotificationActionButtonInfo.java

```diff
@@ -0,0 +1,73 @@
+package org.commcare.views.notifications;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+
+import java.util.Date;
+
+public class NotificationActionButtonInfo implements Parcelable {
+    private final String text;
+    private final ButtonAction action;
+
+    public enum ButtonAction {
+        NONE,
+        LAUNCH_DATE_SETTINGS
+    }
+
+    @Override
+    public int describeContents() {
+        return 0;
+    }
+
+    @Override
+    public void writeToParcel(Parcel dest, int flags) {
+        String buttonText = getButtonText();
+        String buttonAction = getButtonAction().name();
+        dest.writeStringArray(new String[]{buttonText, buttonAction});
```

Comment:
I think a cleaner way to do this is without arrays - 
````
dest.writeString(text)
dest.writeString(action.name) 
````
and then reading it back from source the same way - 
````
new NotificationActionButtonInfo(source.readString(), source.readString());
````

Reply:
Agreed